packages/runtime/runtime-utils/src/utils.ts

```diff
@@ -4,6 +4,25 @@
  */
 
 import { ISerializedHandle } from "@fluidframework/core-interfaces";
+import { IDocumentAttributes, ISnapshotTree } from "@fluidframework/protocol-definitions";
 
 export const isSerializedHandle = (value: any): value is ISerializedHandle =>
     value?.type === "__fluid_handle__";
+
+/** Reads a blob from storage and parses it from JSON. */
+export type ReadAndParseBlob = <T>(id: string) => Promise<T>;
+
+/**
+ * Fetches the sequence number of the snapshot tree by examining the protocol.
+ * @param tree - snapshot tree to examine
+ * @param readAndParseBlob - function to read blob contents from storage
+ * and parse the result from JSON.
+ */
+ export async function seqFromTree(
+    tree: ISnapshotTree,
+    readAndParseBlob: ReadAndParseBlob,
+): Promise<number> {
+    const attributesHash = tree.trees[".protocol"].blobs.attributes;
+    const attrib = await readAndParseBlob<IDocumentAttributes>(attributesHash);
```

Comment:
Curious, why do we need a callback (readAndParseBlob), and not use function directly here.
I think the intention is to hide storage, and I'm not sure why - I'd rather pass in storage object and call function directly.

Reply:
It's just there's no dependency on that package yet, but we could add it. We need both `readAndParse` from driver-utils and `IDocumentStorageService` from driver-definitions.